packages/flutter/test/widgets/scroll_simulation_test.dart

```diff
@@ -23,4 +23,22 @@ void main() {
     checkInitialConditions(75.0, 614.2093);
     checkInitialConditions(5469.0, 182.114534);
   });
+
+
+  test('ClampingScrollSimulation velocity eventually reaches zero', () {
```

Comment:
Note these new tests fail with the old logic. As noted in the my previous comment, velocity ramps back up near the end of the simulation.